### PR TITLE
Create API for Storing Doctor Specializations.

### PR DESCRIPTION
**Pull request by rushiii3**
#### PR Description
Please include a summary of the changes and the related issue. Describe your changes in detail.

#### Type of Change
- [ ] Bug fix (change which fixes an issue)
- [x] New feature (change which adds functionality)
- [ ] Documentation update

#### Select your work-area
- [ ] Frontend
- [x] Backend
- [ ] Documentation
- [ ] Others

#### Related Issue
Please provide a link to the issue solved if applicable.

#### Add your Work Example
Screenshot 2024-05-31 at 2 38 41 PM
Screenshot 2024-05-31 at 2 50 29 PM
Screenshot 2024-05-31 at 2 50 57 PM
Screenshot 2024-05-31 at 2 55 44 PM
Screenshot 2024-05-31 at 2 57 24 PM

#### Fixes (mention the issue number which this fixes)
#76 

#### Checklist
- [x] I have updated my branch and synced it with the project's 'develop' branch before making this PR.
- [x] I have optimized the file changes.
- [x] I have added a snapshot of my work example.
- [x] I have made a PR to the project's develop branch.

#### Undertaking
- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] I have checked for plagiarism and assure its authenticity.
- [x] I have read and followed the code of conduct for this repository. I understand that violation of this undertaking may have legal consequences.

- [x] I Agree

### CODE REVIEW

**Comment by SB2318** [comment]
Looks Good! Will check, You can move with your next other tasks during this time.

File: backend/controllers/specializationsControllers.js [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+const User = require("../models/UserModel");
+var mongoose = require("mongoose");
+
+module.exports.add = async (req, res) => {
+  try {
+    const { name, contributed_by, doctor_ids } = req.body;
+    if (!name) {
+      return res.status(400).json({ error: "Name is empty" });
+    }
+    const isValidContributed_by =
```
Comment (SB2318) [inline backend/controllers/specializationsControllers.js:11]: No, This is not the functionality, As of now at the time of registration user can post a new specialization, if it is not exists in the database.

So, user_id at that point of time not exist I think

File: backend/controllers/specializationsControllers.js [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+      mongoose.Types.ObjectId.isValid(contributed_by); //true
+    if (!isValidContributed_by) {
+      return res.status(400).json({ error: "Invalid Contributor ID" });
+    }
+
+    // Validate user IDs
+    const contributor = await User.find({
```
Comment (SB2318) [inline backend/controllers/specializationsControllers.js:18]: It is okay, we can also add specialization at the time of update profile or from profile page, but not make this mandatory

File: backend/controllers/specializationsControllers.js [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+      _id: contributed_by,
+      isDoctor: false,
+    });
+    if (!contributor) {
+      return res.status(400).json({ error: "User doesn't exist" });
+    }
+    // Validate if each doctor_id is a valid ObjectId
+    for (const id of doctor_ids) {
```
Comment (SB2318) [inline backend/controllers/specializationsControllers.js:26]: doctor_id will be updated from here, in such a way that with this category the newly generated id will be added, for that newly registered user, specialization API will be called from the **register function** and **update profile** function.

File: backend/controllers/specializationsControllers.js (hunk 1 above)
Comment (rushiii3) [reply backend/controllers/specializationsControllers.js:11]: Okay, i have commented the code for now.

File: backend/controllers/specializationsControllers.js (hunk 2 above)
Comment (rushiii3) [reply backend/controllers/specializationsControllers.js:18]: Okay, i have commented the code for now.

**Review by SB2318 — APPROVED** [review]
Nice Work!

### Work Done:

 [✔️]Add Specialization
 [✔️]Update Specialization
[✔️] Get Specialization (By Category Name or By Category ID)
[✔️]Delete Specialization

**Pull request merged**